manipulation/src/torch/csrc/jit/type.h

```diff
@@ -11,124 +11,46 @@
 
 namespace torch { namespace jit {
 
-#define TH_FORALL_TYPES(_) \
-_(TensorType) \
-_(HandleType)
+struct Dimension {
+  Dimension() = delete;
```

Comment:
Deleting this is done implicitly since you've defined a non-trivial constructor below, right?

Reply:
++